crates/attestation/src/crypto/crl.rs

```diff
@@ -0,0 +1,519 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::crypto::{SubjectPublicKeyInfoExt, TbsCertificateExt};
+
+use anyhow::{bail, ensure, Result};
+use const_oid::db::DB;
+use const_oid::ObjectIdentifier;
+use der::asn1::UIntRef;
+use der::{Encode, Sequence};
+use x509::crl::CertificateList;
+use x509::name::Name;
+use x509::PkiPath;
+
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct CrlListEntry<'a> {
+    pub url: String,
+    pub crl: CertificateList<'a>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct CrlList<'a> {
+    pub crls: Vec<CrlListEntry<'a>>,
+}
+
+impl<'a> CrlList<'a> {
+    #[allow(unused)]
+    fn get_crl_by_url(&self, url: &str) -> Option<&CertificateList> {
+        for pair in &self.crls {
+            if pair.url == url {
+                return Some(&pair.crl);
+            }
+        }
+        None
+    }
+
+    #[allow(unused)]
+    fn get_crl_by_issuer(&self, issuer: &Name) -> Option<&CertificateList> {
+        for pair in &self.crls {
+            if &pair.crl.tbs_cert_list.issuer == issuer {
+                println!(
+                    "Returning CRL {} for issuer {}",
+                    pair.crl.tbs_cert_list.issuer, issuer
+                );
+                return Some(&pair.crl);
+            }
+        }
+        None
+    }
+
+    #[allow(unused)]
+    fn is_revoked(&self, issuer: &Name, serial: &UIntRef) -> anyhow::Result<bool> {
+        if let Some(crl) = self.get_crl_by_issuer(issuer) {
+            if let Some(revocations) = crl.tbs_cert_list.revoked_certificates.as_ref() {
+                if revocations.iter().any(|r| r.serial_number == *serial) {
+                    return Ok(true);
+                }
+            }
```

Comment:
We might want to return an `Ok(false)` here (before this closing bracket) since only if we managed to find a CRL for this issuer, and the list of revoked certs is valid, *and* the serial wasn't in the list, should we return an `Ok(false)` I believe.

Reply:
This function was removed.